src/js/popup.js

```diff
@@ -54,38 +63,47 @@
 
   function removeAlert(evt){
     var item = $(evt.currentTarget);
+    removeAlertFromStorage(item);
+    removeAlertFromList(item);
+  }
+
+  function removeAlertFromStorage(item){
     var value = item.attr('data-usd-value');
     chromeService.storage.removeAlert(value);
+  }
+
+  function removeAlertFromList(item){
     item.remove();
-    if(!$('li', alertList).length)
-      alertListContainer.addClass('is-hidden');
+    if(wasLastItemRemoved())
+      hideAlertListContainer();
   }
 
-  function getNowValue() {
-    usdValueService.getCurrentValue()
-      .then(onGetUSDCurrentValueSuccess);
+  function wasLastItemRemoved(){
+    return $('li', alertList).length === 0;
   }
 
-  function onGetUSDCurrentValueSuccess(response){
-    setupUSDCurrentValue(response.dolar.cotacao);
-    setupUSDCurrentValueVariation(response.dolar.variacao);
+  function hideAlertListContainer(){
+    alertListContainer.addClass('is-hidden');
   }
 
-  function setupUSDCurrentValue(dolarValue){
-    var value = dolarValue.substring(0,4);
-    nowValue.text(value);
+  function getCurrentUSDValue() {
+    currencyService.getCurrentUSDValue()
+      .then(onGetCurrentUSDValueSuccess);
```

Comment:
What if we add a `.fail()` here if we are unable for some reason to fetch the current value from the API? Maybe calling a function that will update the `Loading...` text with something like `Failed to fetch value`, so the user will not be waiting for the infinite `Loading...`